File: frontend/next-generation-ticketing-system/src/Service/UserService.ts
import axios from "axios";

const USER_API_BASE_URL = "http://localhost:8080/api/users";

class UserService {
  signup(user: any) {
    return axios.post(USER_API_BASE_URL + "/signup", user);
  }

  login(user: any) {
    return axios.post(USER_API_BASE_URL + "/login", user);
  }

  findUserByEmail(email: string) {
    return axios.get(
      USER_API_BASE_URL + "/search/findUserByEmail?email=" + email
    );
  }

  findByUser(user_id: any) {
    return axios.get(`http://localhost:8080/api/users/${user_id}/auths`);
  }

  // deleteUser(user_id: any) {
  //   return axios.delete(`http://localhost:8080/api/users/` + user_id);
  // }

  getUserById(user_id: any) {
    return axios.post(USER_API_BASE_URL + "/getUser", user_id);
  }

  totalUser() {
    return axios.get(USER_API_BASE_URL + "/totalUser");
  }
}

export default new UserService();

File: backend/ngts/src/main/java/com/three/ngts/Service/UserService.java
package com.three.ngts.Service;

import com.three.ngts.Entity.User;
import com.three.ngts.Repo.UserRepo;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

@CrossOrigin(origins = "http://localhost:3000")
@RestController
@RequestMapping("/api/")
public class UserService {

    @Autowired
    private UserRepo userRepo;

    @PostMapping("/users/getUser")
    public User getUser(@RequestBody String userId) {
        return userRepo.findByUserId(Long.parseLong(userId));
    }

    @PostMapping("/users/signup")
    public User signUp(@RequestBody User user) {
        return userRepo.save(user);
    }

    @PostMapping("/users/login")
    public boolean login(@RequestBody User user) {
        try {
            User newUser = userRepo.findUserByEmail(user.getEmail());
            return newUser.getEmail().equals(user.getEmail()) && newUser.getPassword().equals(user.getPassword());
        } catch (Exception e) {
            return false;
        }
    }

    // @DeleteMapping("/users/{userId}")
    // public ResponseEntity<String> deleteUser(@PathVariable Long userId) {
    // Optional<User> optionalUser = userRepo.findById(userId);
    //
    // if (optionalUser.isPresent()) {
    // userRepo.deleteById(userId);
    // return ResponseEntity.ok("User deleted successfully");
    // } else {
    // return ResponseEntity.status(HttpStatus.NOT_FOUND).body("User not found");
    // }
    // }

    @GetMapping("/users/totalUser")
    public Integer totalUser() {
        return userRepo.findAll().size();
    }
}
